bcda/cclf/cclf.go

```diff
@@ -67,22 +67,27 @@ func importCCLF0(fileMetadata *cclfFileMetadata) (map[string]cclfFileValidator,
 	)
 
 	var validator map[string]cclfFileValidator
-	for i, f := range r.File {
-		fmt.Printf("Reading file #%d from archive %s.\n", i, fileMetadata)
-		log.Infof("Reading file #%d from archive %s", i, fileMetadata)
-		if err := validateFileName(f.Name); err != nil {
-			fmt.Printf("Unknown file name when validating file: %s.\n", f.Name)
-			err = errors.Wrapf(err, "unknown file name when validating file: %s.\n", f.Name)
-			log.Error(err)
-			return nil, err
+	var rawFile *zip.File
+
+	for _, f := range r.File {
+		// needed for split files, otherwise this could have been an '==' check
+		if strings.Contains(f.Name, fileMetadata.name){
+			rawFile = f
+			fmt.Printf("Reading file %s from archive %s.\n", fileMetadata.name, fileMetadata.filePath)
+			log.Infof("Reading file %s from archive %s", fileMetadata.name, fileMetadata.filePath)
 		}
-		if err = parseTimestamp(fileMetadata, f.Name); err != nil {
+
+		if rawFile == nil {
+			fmt.Printf("File %s not found in archive %s.\n", fileMetadata.name, fileMetadata.filePath)
+			err = errors.Wrapf(err, "file %s not found in archive %s", fileMetadata.name, fileMetadata.filePath)
+			log.Error(err)
 			return nil, err
 		}
-		rc, err := f.Open()
+
+		rc, err := rawFile.Open()
```

Comment:
Trying to understand what this is doing... is it opening a zip (i.e., CCLF0) within the top-level zip?  Or is this how we are intending to access the plaintext file within the top-level zip?  Remember, according to the tech spec (and according to the files that have been distributed via ACO-OS), only the top-level zip is in zip format and all files within the zip (including CCLF0 and CCLF8) are in plaintext format.

Reply:
this is opening up the actual file